libcudacxx/include/cuda/std/__atomic/api/atomic_crtp.h

```diff
@@ -0,0 +1,42 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef __LIBCUDACXX___ATOMIC_API_ATOMIC_CRTP_H
+#define __LIBCUDACXX___ATOMIC_API_ATOMIC_CRTP_H
+
+#include <cuda/std/detail/__config>
+
+_LIBCUDACXX_BEGIN_NAMESPACE_STD
+
+// __atomic_crtp_accessor defines a way to statically fetch the atomic storage object
+// which owns the stored atomic.
+template <typename _Impl, typename _Sto>
+struct __atomic_crtp_accessor {
```

Comment:
Just for me, didnt we want to get away from the crtp accessor?

Reply:
Done